src/lib/auth/README.md

```diff
@@ -0,0 +1,44 @@
+## Authentication Library
+Initialize authentication library (authenticator). A strategy will be prepared based on the authentication type.
+
+**The library is promised based, all exposed methods return promises.**
+
+The module acts like a singleton. Both the strategy and the authentication options are saved in
+the package's object which allows us to only initialize the library once. Meaning that after we
+initialize the library it will be shared throughout the entire codebase without the need of
+re-initialization.
+
+Currently implemented strategies are:
+ - `wpcom-oauth` - regular wpcom oAuth process using `wpcom-oauth-cors` package
+ - `wpcom-oauth-by-token` - regular wpcom oAuth based on given token using `wpcom-xhr-request`
+ - `wpcom-proxy-iframe` - proxy iframe wpcom authentication using Calypso's wpcom object
+
+ ## API
+ `init( settings )` - Initialize authentication library (authenticator). A strategy will be prepared based on the authentication type
```

Comment:
A nitpick here is that are asking users to configure twice the auth: first with the `type`, and then in the `options` object. How about simplifying the settings to allow receiving either:

- a `string`: the oauth token => we use the `wpcom-oauth-token` strategy
- an `object`: the proxy object => we use the `wpcom-proxy-iframe` strategy

Reply:
Another option will be to always pass an object, but depending on the key used (`token` or `proxy`) we'd know which strategy to use. For example, for `wpcom-oauth-token`:

    Happychat.open( {
        authentication: { token: '<a token here >'},
       ...
    });

and for `wpcom-proxy-iframe`:

    Happychat.open( {
        authentication: { proxy: '<a token here >'},
       ...
    });

I propose this simplification (or something on these lines) because we need to think about the context where this will end up: the Happychat [config options](https://github.com/Automattic/happychat-client/blob/master/docs/API.md#init-settings). It already has quite a few options, and if every option has a lot of params, the Happychat config is going to be horribly complicated.

Reply:
I wanted to keep the type also in order to have a more generic authentication. Right now we do it via WPCOM so token and proxy might make sense but I think we should keep the door open for other strategies to be implemented. In that scenario having optional parameters like, token and proxy could become confusing if they are not tied to a specific authentication strategy.